utoipa/src/openapi/schema.rs

```diff
@@ -236,6 +236,59 @@ pub enum Schema {
     AllOf(AllOf),
 }
 
+impl<'de> Deserialize<'de> for Schema {
+    fn deserialize<D>(deserializer: D) -> Result<Schema, D::Error>
+    where
+        D: Deserializer<'de>
+    {
+        let v: Value = serde::Deserialize::deserialize(deserializer)?;
+        match &v {
+            Value::Object(ob) => {
+                if ob.contains_key("type") {
+                    match ob.get("type").unwrap() {
+                        Value::String(s) => {
+                            if s == "array" {
+                                let ro: Result<Array, serde_json::Error> = serde::Deserialize::deserialize(v);
+                                match ro {
+                                    Ok(o) => Ok(Schema::Array(o)),
+                                    Err(msg) => panic!("{}", msg)
+                                }
+                            } else if s == "oneof" {
+                                let ro: Result<OneOf, serde_json::Error> = serde::Deserialize::deserialize(v);
+                                match ro {
+                                    Ok(o) => Ok(Schema::OneOf(o)),
+                                    Err(msg) => panic!("{}", msg)
+                                }
+                            } else if s == "allof" {
+                                let ro: Result<AllOf, serde_json::Error> = serde::Deserialize::deserialize(v);
+                                match ro {
+                                    Ok(o) => Ok(Schema::AllOf(o)),
+                                    Err(msg) => panic!("{}", msg)
+                                }
+                            } else {
+                                // per default it is tried to deserialize as an object
+                                let ro: Result<Object, serde_json::Error> = serde::Deserialize::deserialize(v);
+                                match ro {
+                                    Ok(o) => Ok(Schema::Object(o)),
+                                    Err(msg) => panic!("{}", msg)
+                                }
```

Comment:
For `array` and to the `object` this _type_ field check might work just fine. But for `oneOf` and for `allOf` there is no type field to check. Instead they just are formed like so:
```yaml
oneOf:
  - #/schemas/components/Foobar
---
allOf:
  - #/schemas/components/Foobar
```
Probably there is something more sophisticated that need to be done in order to cover these as well.

Reply:
Yes, but nothing to complicated. Maybe I will have some time tomorrow to take a look at this.

Reply:
That would be awesome. Thank you for your time investment with this library :slightly_smiling_face: I am also little by little getting back to the routine since the personal things are settled more or less.

Reply:
Hey so I have some changes that fix this now on my personal verison, will push it soon … but now I am a bit puzzled about the `OneOf` and `AllOf` structures of Utoipa. if `oneOf` or `allOf` are supposed to be serialized/deserialized to/from an array, that is not compatible with the other keys present in the Struct like `description` and so forth.